multicluster/controllers/multicluster/member/clusterset_controller.go

```diff
@@ -90,57 +92,69 @@ func NewMemberClusterSetReconciler(client client.Client,
 func (r *MemberClusterSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	clusterSet := &mcv1alpha2.ClusterSet{}
 	err := r.Get(ctx, req.NamespacedName, clusterSet)
-	r.commonAreaLock.Lock()
-	defer r.commonAreaLock.Unlock()
-	if err != nil {
-		if !apierrors.IsNotFound(err) {
-			return ctrl.Result{}, err
-		}
-		klog.InfoS("Received ClusterSet delete", "clusterset", req.NamespacedName)
-		if r.remoteCommonArea != nil {
-			if err := r.deleteMemberClusterAnnounce(ctx); err != nil {
-				// MemberClusterAnnounce could be kept in the leader cluster, if antrea-mc-controller crashes after the failure.
-				// Leader cluster will delete the stale MemberClusterAnnounce with a garbage collection mechanism in this case.
-				return ctrl.Result{}, fmt.Errorf("failed to delete MemberClusterAnnounce in the leader cluster: %v", err)
+	processClusterSet := func() error {
+		r.commonAreaLock.Lock()
+		defer r.commonAreaLock.Unlock()
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return err
 			}
-			r.remoteCommonArea.Stop()
-			r.remoteCommonArea = nil
-			r.clusterSetConfig = nil
-			r.clusterID = common.InvalidClusterID
-			r.clusterSetID = common.InvalidClusterSetID
+			klog.InfoS("Received ClusterSet delete", "clusterset", req.NamespacedName)
+			if r.remoteCommonArea != nil {
+				if err := r.deleteMemberClusterAnnounce(ctx); err != nil {
```

Comment:
Probably add a comment saying ResourceExports will be deleted by the leader cluster.

Reply:
Comment updated.